pkg/common/auth_credentials/auth_credentials.go

```diff
@@ -38,14 +56,29 @@ func (c *AuthCredential) GetCredentialsFromReq(httpReq *envoyServiceAuthV3.Attri
 	case inQuery:
 		return getCredFromQuery(httpReq.GetPath(), c.KeySelector)
 	default:
-		return "", fmt.Errorf("the credential location is not supported")
+		return "", fmt.Errorf(credentialLocationNotSupported)
+	}
+}
+
+func (c *AuthCredential) GetCredentialsFromCluster(ctx context.Context, labelSelectors map[string]string) ([]string, error) {
+	var matchingLabels client.MatchingLabels = labelSelectors
+	var secretList = &v1.SecretList{}
+	if err := c.k8sClient.List(ctx, secretList, matchingLabels); err != nil {
+		return nil, err
 	}
+	var parsedSecrets []string
+
+	for _, secret := range secretList.Items {
+		parsedSecrets = append(parsedSecrets, string(secret.Data[c.KeySelector]))
```

Comment:
Wee need to highlight this in the instructions somehow. BTW, the key of the k8s secret is case sensitive.

Reply:
Yeah, you are right, this is kind of key to make it work

Reply:
It's case sensitive, but we are forced to make it lowercase when working with headers. Envoy stores them lowercase.